### ISSUE TITLE
app crashes if debugging mode is enabled

### ISSUE THREAD

**Issue by Yandamuri** [software-mansion/react-native-reanimated#1674]
## Description
I am using `react-native-reanimated` as a dependency of `react-native-draggable-flatlist`. Since I started using `reanimated`,  app crashes if debugging mode is enabled. Once I uninstall `reanimated` package then app runs in debugging mode perfectly.

### Screenshots
![Screenshot_1611898820](https://user-images.githubusercontent.com/31983789/106236457-ebd5ce80-6222-11eb-86cb-64078b73f671.png)

## Steps To Reproduce

1. Just install the  `react-native-reanimated`  as per docs and then enable debugging mode. You can encounter this error

### Expected behavior
debugging mode should work as usual

### Actual behavior
app crashes if debugging mode is enabled

## Snack or minimal code example
No specific code is required. Just installing the package is sufficient.

## Package versions
- React: "16.13.1"
- React Native: "0.63.2"
- React Native Reanimated:"2.0.0-rc.2"
- NodeJS: v10.16.2

**Comment by kaioduarte** [comment]
It seems they don't support v2 yet, install v1 instead.

**Comment by Yandamuri** [comment]
@kaioduarte what do mean by "they" here? Can you please tell me clearly?

**Comment by kaioduarte** [comment]
@Yandamuri the maintainers of `react-native-draggable-flatlist`, there is an open issue asking for support reanimated v2.

**Comment by jakub-gonet** [comment]
Issue link: https://github.com/computerjazz/react-native-draggable-flatlist/issues/247

There's not much we can do right now about that. Enabling turbomodules disables debugging by chrome, try using Flipper instead.

**Comment by Yandamuri** [comment]
@jakub-gonet Thank you for your response and suggestion. 

I m not sure if it is appropriate to ask here. Can you please have a look at this [Stack Overflow](https://stackoverflow.com/questions/65988564/flippersdebugging-platform-logs-data-is-collapsed-by-default) question regarding Flipper, if you have time?

**Comment by wodCZ** [comment]
@jakub-gonet is there something we could do to let user (developer) know that the debug mode causes crash?

I've spent longer than comfortable amount of time trying to find why the app is crashing with `No source URL loaded, have you initialised the instance`.

Found this thanks to mention in https://github.com/expo/expo/pull/10383

Maybe the `ReanimatedJSIModulePackage` could check whether it's running in remote debugging mode and log a message that the app can't be run in such way before crashing?

I know it's mentioned in known limitations that the debug mode is not supported, it's just easily overlooked (especially if someone else in the team does the setup for you).

**Comment by tuneerclixlogix** [comment]
I am also facing same issue of No source URL loaded, have you initialised the instance in debugging mode. Has anyone found any solution yet with the react native cli ?

**Comment by newfylox** [comment]
I've been looking for hours about this bug and now I found this issue on Github. I was using a new ejected Expo app to notice that **react-native-reanimated** was installed with v2. Using Flipper instead of Chrome DevTools or removing this package solved the problem.

I am using `yarn start` instead of `expo start`, probably that's why.

There should definitely have an another warning message to tell people that it can be the problem.

**Comment by on3iro** [comment]
Hi, I am also currently having this issue (just using `react-native-reanimated`). I do not quite understand what the proposed fix for this is (apart from uninstalling `reanimated`). We need the debugger mode for various reasons, so any help would be appreciated :)

**Comment by yakmoon** [comment]
Apparently there is no solution for this atm. it's very frustrating to code without the debugger. and I can not uninstall reanimated just for this.

**Comment by wodCZ** [comment]
This is not related just to reanimated and the debugger support is not a thing reanimated can "fix", since it's because of react native architecture (or better, turbo modules).
While it's annoying that the app just crashes when you enable remote debugging, it doesn't seem that reanimated can do anything more than adding a huge warning box to the [installation instructions](https://docs.swmansion.com/react-native-reanimated/docs/2.0.0/installation/#installing-the-package).

- Like the box there says, you can **use Flipper to debug your app**. 
- Some tips on how to debug were also mentioned in [RN's Turbo Modules topic](https://github.com/react-native-community/discussions-and-proposals/issues/40#issuecomment-813684618). You can try using Safari to debug iOS app.
- There are also special [Hermes debugging instructions](https://reactnative.dev/docs/hermes#debugging-js-on-hermes-using-google-chromes-devtools).
- With expo it's more complicated as you can find [in their documentation](https://docs.expo.dev/versions/latest/sdk/reanimated/#:~:text=The%20new%20APIs,in%20Reanimated%201.). You can follow the [feature request](https://expo.canny.io/feature-requests/p/integrate-flipper) and try [this workaround](https://twitter.com/Baconbrix/status/1412921581542658049?s=20) in the meanwhile.
- You can [join the discussion](https://github.com/react-native-community/discussions-and-proposals/issues/206) at react native's proposal repo and propose a solution. Or just subscribe.

**Comment by farshidshahmoradi1996** [comment]
same problem .

**Comment by valery-lavrik** [comment]
a similar problem.
Do I understand correctly that react-native should solve the problem???

**Comment by ashishprasad2163** [comment]
++

**Comment by valery-lavrik** [comment]
it seems there is a commit with a solution to the problem. It remains to wait for the new release and you can check again

https://github.com/software-mansion/react-native-reanimated/pull/2795/files

**Comment by ashishprasad2163** [comment]
Thanks @valery-lavrik.

**Comment by ashishprasad2163** [comment]
@valery-lavrik 
Project/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/ReanimatedProxyPackage.java
I'm finding only this file inside the specified path in the commit.
And looks like something is missing from the class. version:"react-native-reanimated": "^2.3.1"
Am I doing some mistake installing this package?
```
package com.swmansion.reanimated;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;

import java.util.Arrays;
import java.util.List;

import com.swmansion.reanimated.ReanimatedPackage;

public class ReanimatedPackage implements ReactPackage {
  // dummy package
}

```

**Comment by valery-lavrik** [comment]
Fixed in release   https://www.npmjs.com/package/react-native-reanimated/v/2.4.0

**Comment by ashishprasad2163** [comment]
@valery-lavrik .[This](https://github.com/software-mansion/react-native-reanimated/issues/1674#issuecomment-1031108848) worked for me. Thanks a ton!

After installing react-native-reanimated 2.4.0, the application stopped crashing on debugging mode enabled.

After launching, the application became unresponsive with error: 
Invariant Violation: Calling synchronous methods on native modules is not supported in Chrome.

I am on RN 0.65.0, so I found a link that states a temporary [fix](https://stackoverflow.com/questions/61067004/invariant-violation-calling-synchronous-methods-on-native-modules-is-not-suppor) for the error.
[Link](https://stackoverflow.com/questions/61067004/invariant-violation-calling-synchronous-methods-on-native-modules-is-not-suppor)

```
callNativeSyncHook(
    moduleID: number,
    methodID: number,
    params: any[],
    onFail: ?Function,
    onSucc: ?Function,
  ): any {
    const isDebuggingEnabled = (typeof atob !== 'undefined');
    this.processCallbacks(moduleID, methodID, params, onFail, onSucc);
    if(!isDebuggingEnabled)
    {
      return global.nativeCallSyncHook(moduleID, methodID, params);
    }
  }
```
make these changes in the given file: node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js

After making these changes, I'm to run the application in debug mode without any crashes.

Looking for some permanent fix.

**Comment by sanjeevdhavala** [comment]
I was on react-native-reanimated version 2.3.1 and after upgrading it to 2.4.1, it worked. Thank you @valery-lavrik

**Comment by uzuntonev** [comment]
It was fixed for me after being updated to ` react-native-reanimated 2.4.0`

**Comment by anutting** [comment]
It looks like this was only fixed on iOS, any word on a fix for Android?

**Comment by pierroo** [comment]
I can confirm this doesn't work on android, I am also getting the error "Invariant Violation: Calling synchronous methods on native modules is not supported in Chrome."
Using latest version 2.5.0

**Comment by fadi-quader-mox** [comment]
the issue still exists on Android

**Comment by valery-lavrik** [comment]
> the issue still exists on with version 2.6.0

https://docs.swmansion.com/react-native-reanimated/docs/

![image](https://user-images.githubusercontent.com/8047999/162672537-275d39bd-52a7-40c2-8d46-38e6ba367acf.png)

**Comment by fadi-quader-mox** [comment]
@valery-lavrik it should be fixed in version `2.4.0` https://github.com/software-mansion/react-native-reanimated/releases/tag/2.4.0
